string state_tag
int32 action_number
